### PR TITLE
feature(format): number formatting

### PR DESCRIPTION
**Pull request by lucasrodes**
Format numbers

```python
>>> from owid.datautils.format import format_number
>>> format_number("1,000")
1000
>>> format_number("1 000 234")
1000234
>>> format_number("1 million 1 hundred")
1000100
```

### CODE REVIEW

File: owid/datautils/format/numbers.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        # Force string
+        number = num_to_str(number)
+        # Remove multiple spaces
+        number = remove_multiple_whitespaces(number)
+        return number
+
+    def clean(self) -> int:
+        """Clean number."""
+        if self.number.isnumeric():  # already a number
+            return int(self.number)
```
Comment (pabloarosado) [inline owid/datautils/format/numbers.py:36]: What if the number has decimals?

File: owid/datautils/format/numbers.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            raise ValueError(f"Invalid number format {self.number}")
+
+
+class NumberWithSeparators:
+    """Class for numbers with separators.
+
+    Accepted separators are: '.', ',' and ' '.
+    """
+
+    accepted_separators = [r"\.", ",", r"\s"]
```
Comment (pabloarosado) [inline owid/datautils/format/numbers.py:51]: What if the number has separators and a decimal point (e.g. "1,234.567")?

File: owid/datautils/format/numbers.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    str
+        String without duplicated whitespaces.
+    """
+    return re.sub(r"\s+", " ", text)
+
+
+def format_number(number: Union[int, str]) -> int:
+    """Format number.
+
+    Only supports integer conversion.
```
Comment (pabloarosado) [inline owid/datautils/format/numbers.py:286]: Ah right, it's only meant for integers. If this is not expected to be generalised any time soon, maybe the name of the function should reflect it, e.g. `clean_integer`.

File: tests/format/test_numbers.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    def test_match_numeric_words_ok(self):
+        numbers = {
+            "2 million": {
+                "million": "2",
+                "ten_thousand": 0,
+                "thousand": 0,
+                "hundred": 0,
+                "one": 0,
+            },
+            "2 million 1": {
```
Comment (pabloarosado) [inline tests/format/test_numbers.py:78]: Shouldn't it be "2 million and 1" or even better "Two million and one"? I don't know if anyone would write numbers like "2 million 1".

**Review by pabloarosado — APPROVED** [review]
I don't think this logic can be applied to many use cases, but I suppose in the COVID repos you have found some examples, so it may be useful sometimes. I have the impression that the logic may be sometimes too smart and sometimes not smart enough, making it in practice unreliable. In particular, the fact that decimal points are not allowed feels like a big downside. The only use case I could think of (FAOSTAT) uses both separator and decimal points (see [this example](https://github.com/owid/etl/blob/a580e0b093225f4c0110f5d16681c4b95d434bde/etl/steps/data/garden/faostat/2022-05-17/shared.py#L345)).
My main suggestion would be to clarify (in docstrings and possibly function names) the limitations of the functions.

File: tests/format/test_numbers.py (hunk 4 above)
Comment (lucasrodes) [reply tests/format/test_numbers.py:78]: I will add a `.replace('and ', '')` so it works when 'and' is used

**Comment by lucasrodes** [comment]
In general, this function is very basic at the moment. I'd suggest clarifying further that it only supports integer cleaning and that it should be used with caution at the moment.

I think we can improve it a lot in the future, but for the time being, I migrated the essential functionalities.

File: owid/datautils/format/numbers.py (hunk 1 above)
Comment (lucasrodes) [reply owid/datautils/format/numbers.py:36]: I've clarified that this only works for Integer values at the moment

File: owid/datautils/format/numbers.py (hunk 2 above)
Comment (lucasrodes) [reply owid/datautils/format/numbers.py:51]: I've clarified that this only works for Integer values at the moment

File: owid/datautils/format/numbers.py (hunk 3 above)
Comment (lucasrodes) [reply owid/datautils/format/numbers.py:292]: I'll leave it as it is now, might add float support in the future.

**Pull request merged**